site/app/controllers/OfficeHoursQueueController.php

```diff
@@ -0,0 +1,217 @@
+            new RedirectResponse($this->core->buildCourseUrl(['office_hours_queue']))
+        );
+    }
+
+    /**
+    * @param
+    * @Route("/{_semester}/{_course}/office_hours_queue/finishHelp", methods={"POST"})
+    * @AccessControl(role="LIMITED_ACCESS_GRADER")
+    * @return Response
+    */
+    public function finishHelpPerson(){
+        if(!isset($_POST['entry_id'])){
+            $this->core->addErrorMessage("Missing entry ID");
+            return Response::RedirectOnlyResponse(
+                new RedirectResponse($this->core->buildCourseUrl(['office_hours_queue']))
+            );
+        }
+        if($this->core->getUser()->accessGrading())
+        $this->core->getQueries()->finishHelpUser($_POST['entry_id'], $this->core->getUser()->getId());
+        return Response::RedirectOnlyResponse(
+            new RedirectResponse($this->core->buildCourseUrl(['office_hours_queue']))
+        );
+    }
+
+    /**
+    * @param
+    * @Route("/{_semester}/{_course}/office_hours_queue/toggle", methods={"POST"})
+    * @AccessControl(role="LIMITED_ACCESS_GRADER")
+    * @return Response
+    */
+    public function toggleQueue(){
+        if(!isset($_POST['queue_open'])){
+            $this->core->addErrorMessage("Missing queue status");
+            return Response::RedirectOnlyResponse(
+                new RedirectResponse($this->core->buildCourseUrl(['office_hours_queue']))
+            );
+        }
+        if(!$this->core->getUser()->accessGrading()){
+            return Response::RedirectOnlyResponse(
+                new RedirectResponse($this->core->buildCourseUrl(['office_hours_queue']))
+            );
+        }
+        if($_POST['queue_open'] == "Open Queue"){
+            $this->core->getQueries()->openQueue();
+        } else{
+            $this->core->getQueries()->closeQueue();
+        }
+        return Response::RedirectOnlyResponse(
+            new RedirectResponse($this->core->buildCourseUrl(['office_hours_queue']))
+        );
+    }
+
+    /**
+    * @param
+    * @Route("/{_semester}/{_course}/office_hours_queue/remove", methods={"POST"})
+    * @return Response
+    */
+    public function removePerson(){
+        if(!isset($_POST['entry_id'])){
+            $this->core->addErrorMessage("Missing entry ID");
+            return Response::RedirectOnlyResponse(
+                new RedirectResponse($this->core->buildCourseUrl(['office_hours_queue']))
+            );
+        }
+        if(!$this->core->getUser()->accessGrading()){
+            if($this->core->getUser()->getId() != $this->core->getQueries()->getUserIdFromQueueSlot($_POST['entry_id'])){
+                $this->core->addErrorMessage("Permission denied to remove that person");
+                return Response::RedirectOnlyResponse(
+                    new RedirectResponse($this->core->buildCourseUrl(['office_hours_queue']))
+                );
+            }
+            $this->core->addSuccessMessage("Removed from queue");
```

Comment:
this should probably be moved outside the if statement so that teaching staff also get this success message